slither/detectors/variables/possible_const_state_variables.py

```diff
@@ -71,54 +75,66 @@ def _constant_initial_expression(self, v: Variable) -> bool:
         if not v.expression:
             return True
 
+        # B b = new B(); b cannot be constant, so filter out and recommend it be immutable
+        if isinstance(v.expression, CallExpression) and isinstance(
+            v.expression.called, NewContract
+        ):
+            return False
+
         export = ExportValues(v.expression)
         values = export.result()
         if not values:
             return True
-        if all((val in self.valid_solidity_function or _is_constant_var(val) for val in values)):
-            return True
-        return False
 
-    def _detect(self) -> List[Output]:
-        """Detect state variables that could be const"""
-        results = []
+        return all((val in self.valid_solidity_function or _is_constant_var(val) for val in values))
 
-        all_variables_l = [c.state_variables for c in self.compilation_unit.contracts]
-        all_variables: Set[StateVariable] = {
-            item for sublist in all_variables_l for item in sublist
+    def _detect(self) -> List[Output]:
+        """Detect state variables that could be constant or immutable"""
+        results = {}
+
+        variables = []
+        functions = []
+        for c in self.compilation_unit.contracts:
+            if is_openzeppelin(c):
```

Comment:
Why removing openzepellin? 

Maybe we should iterate over `compilation_unit.contracts_derived` here, so that we only consider the most derived contracts, and we avoid FP on potential constant variable that are written in a derived contract